sdks/go/pkg/beam/core/util/reflectx/call.go

```diff
@@ -20,44 +20,50 @@ import (
 	"reflect"
 	"sync"
 
+	"fmt"
+	"runtime/debug"
+
 	"github.com/apache/beam/sdks/go/pkg/beam/log"
 )
 
 //go:generate specialize --input=calls.tmpl
 
-// Caller is an untyped function call interface. This indirection allows
-// us to avoid reflection call overhead for certain types.
-type Caller interface {
+// Func represents a callable untyped function. This indirection allows
+// us to avoid reflection call overhead for certain types as well as
+// faster dynamic function implementations.
+type Func interface {
+	// Name returns the name of the function.
+	Name() string
 	// Type returns the type.
 	Type() reflect.Type
 	// Call invokes the implicit fn with arguments.
 	Call(args []interface{}) []interface{}
 }
 
 var (
-	callers   = make(map[string]func(interface{}) Caller)
-	callersMu sync.Mutex
+	funcs   = make(map[string]func(interface{}) Func)
+	funcsMu sync.Mutex
 )
 
-// RegisterCaller registers an custom caller factory for the given type, such as
-// "func(int)bool". If multiple caller factories are registered for the same type,
+// RegisterFunc registers an custom reflectFunc factory for the given type, such as
```

Comment:
<!--new_thread; commit:9c2390201ba3ecb080c84c7963396368ee720ae0; resolved:0-->
reflectFunc -> reflectx.Func ?
Or maybe just Func? it seems appropriate to drop the package specifier for local Exported types.

Reply:
Done